widgets/src/components/Attachments/AttachmentsContainer.js

```diff
@@ -150,9 +155,50 @@ class AttachmentsContainer extends React.Component {
     this.setState({ notification: {} });
   };
 
+  handleDownload = item => async e => {
+    const { connection } = this.state;
+    const { taskId } = this.props;
+
+    e.preventDefault();
+
+    try {
+      this.setState({ loading: true });
+      const response = await downloadAttachments(connection, taskId, item.id);
+
+      // read stream
+      const reader = response.body.getReader();
+      const chunks = [];
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
```

Comment:
Can't we use the async iterators? E.g.
```
async function getResponseSize(url) {
  const response = await fetch(url);
  let total = 0;

  for await (const chunk of response.body) {
    total += chunk.length;
  }
  return total;
}
```

Reply:
Talking with Martynas, he suggested a solution from this article:
https://jakearchibald.com/2017/async-iterators-and-generators/#making-streams-iterate

but as it doesn't solve the main issue that is using the `while (true)`, we agreed to keep the implementation for now, and revisit it later.